dds/DCPS/transport/framework/TransportRegistry.h

```diff
@@ -157,15 +167,30 @@ class OpenDDS_Dcps_Export TransportRegistry {
     OPENDDS_STRING transport_template_name;
     OPENDDS_STRING config_name;
     bool instantiate_per_participant;
-    OPENDDS_MAP(OPENDDS_STRING, OPENDDS_STRING) customizations;
-    OPENDDS_MAP(OPENDDS_STRING, OPENDDS_STRING) transport_info;
+    ValueMap customizations;
+    ValueMap transport_info;
   };
 
   OPENDDS_VECTOR(TransportTemplate) transport_templates_;
 
   bool get_transport_template_info(const ACE_TString& config_name, TransportTemplate& inst);
 
-  bool has_transport_template() const;
+  bool has_transport_templates() const;
+
+  struct TransportEntry
+  {
+    OPENDDS_STRING transport_name;
+    OPENDDS_STRING config_name;
+    ValueMap transport_info;
+  };
+
+  OPENDDS_VECTOR(TransportEntry) transports_;
+
+  bool get_transport_info(const ACE_TString& config_name, TransportEntry& inst);
+
+  bool has_transports() const;
+
+  OPENDDS_MAP(DDS::DomainId_t, OPENDDS_STRING) domain_config_map_;
```

Comment:
Coming back to this now, since these are config names, maybe this structure ought to hold ACE_TStrings instead of OPENDDS_STRINGs?